app/js/data/pool.js

```
@@ -72,6 +74,70 @@ class Pool {
+    let poolIndex = 0;
+    for (let cardIndex = 0; cardIndex < 25; cardIndex++) {
+      if (cardIndex === 12) continue; // Skip FREE space
+
+      const cardEntry = cardSpaces[cardIndex];
+
+      // Find matching entry in contents by text
+      const matchingReservation = this.contents.find(reservation =>
+        reservation.entry.text === cardEntry.text
+      );
+
+      if (matchingReservation) {
+        this.spaces[poolIndex] = matchingReservation.entry;
+        matchingReservation.entry.used = true;
+      } else {
+        console.warn(`Could not find matching entry for: ${cardEntry.text} - entry may have been removed from dataset`);
+        // Continue without this entry, it will be undefined in this.spaces[poolIndex]
+      }
+      poolIndex++;
+    }
+  }
+
+  // Reroll a specific pool space given a slot number and update the pool's spaces array
+  rerollPoolSpace(slot) {
+    if (slot < 0 || slot >= 24) {
+      throw new Error('Invalid pool index');
+    }
+
+    const currentEntry = this.spaces[slot];
+
+    // Generate a random starting position
+    const randomSeed = new Uint32Array(1);
+    self.crypto.getRandomValues(randomSeed);
+    const initial_offset = randomSeed[0] % this.contents.length;
+
+    try {
+      const offset = (initial_offset + this.getNextValidReservation(initial_offset)) % (this.contents.length - 1);
+      const newEntry = this.contents[offset].entry;
+
+      // Update pool's spaces array
+      this.spaces[slot] = newEntry;
+      this.spaces[slot].used = true;
+
+      // Handle case where current entry might be undefined (from missing dataset entries)
+      if (currentEntry) {
+        // Set previous entry to unused only after generating a new entry.
+        // This is to prevent regenerating the same entry.
+        currentEntry.used = false;
+      }
+
+      return newEntry;
+    } catch (error) {
+      throw error;
+    }
```

Why catch just to throw again without anything in between? Might as well remove the try/catch wrapping and just let it throw.

---

ah... my brain is too used to thinking in Rust. have fixed it, thanks!